### ISSUE TITLE
SweepLine - Add non-simple/degenerate polygon check

### ISSUE THREAD

**Issue by wnbittle** [dyn4j/dyn4j#75]
In addition to #72 here are some additional examples of SweepLine failing.  All of these are non-simple polygons.  This issue is a request to check the input either before or during processing to produce a better error.

- `new Vector2[] {new Vector2(0.1595990921676319, 0.20158036631684495),new Vector2(0.3627243978540108, -0.2125801642934565),new Vector2(0.4972213824759445, -0.2197501458724339),new Vector2(-0.17530050402164232, -0.10202036313267437)}` 
Exception in thread "main" java.lang.NullPointerException
	at org.dyn4j.geometry.decompose.SweepLine.regular(SweepLine.java:258)

- `new Vector2[] {new Vector2(0.187521000630546, -0.2171227524343904),new Vector2(-0.05418163781638374, -0.4552384293706746),new Vector2(-0.12615265827683775, 0.08842525905551823),new Vector2(-0.4197343412893181, -0.45293439849558936)}`
Exception in thread "main" java.lang.NullPointerException
	at org.dyn4j.geometry.decompose.SweepLine.merge(SweepLine.java:223)

- `new Vector2[] {new Vector2(-0.22574647794211955, 0.3562272754868271),new Vector2(-0.24724056392833493, -0.06552204150010887),new Vector2(0.2551995234048088, -0.4678431592201415),new Vector2(-0.11272047497863902, -0.40936273068655504)}`
Exception in thread "main" java.lang.NullPointerException
	at org.dyn4j.geometry.decompose.SweepLine.split(SweepLine.java:190)

- `new Vector2[] {new Vector2(-0.07792188619765694, 0.10364292899125216),new Vector2(0.2412466770151972, -0.3145214553981004),new Vector2(0.21037640391727175, 0.06289919008100842),new Vector2(0.3079072605141815, -0.20863138522549773)}`
Causes infinite loop

**Comment by wnbittle** [comment]
Here's an example of a degenerate simple polygon with zero area that causes an NPE.  This is from #72 
```
Vector2[] vertices = new Vector2[] {
				new Vector2(70.5, 360.0),
				new Vector2(70.947212,360.89444),
				new Vector2(71.394424,361.78884899999997),
				new Vector2(71.158356,361.316711),
				new Vector2(70.71114299999999,360.422302)
		};
```

**Comment by wnbittle** [comment]
Fixed in commit 1134b731412055bfa1f6e7e7e49d4f6d6874baf4
